assets/src/edit-story/components/panels/textStyle/color.js

```diff
@@ -109,6 +126,46 @@ function ColorControls({ selectedElements, pushUpdate }) {
     [backgroundTextMode, pushUpdate]
   );
 
+  const padding = useCommonObjectValue(
+    selectedElements,
+    'padding',
+    DEFAULT_PADDING
+  );
+
+  const handleBackgroundModeButton = useCallback(
+    (value, mode) => {
+      if (!value) {
+        return;
+      }
+
+      pushUpdate(({ x, y, width, height }) => {
+        let updates = { backgroundTextMode: mode };
+        if (
+          mode === BACKGROUND_TEXT_MODE.FILL ||
+          mode === BACKGROUND_TEXT_MODE.HIGHLIGHT
+        ) {
+          const newPadding = DEFAULT_PADDING_FOR_MODE[mode];
+          const { horizontal, vertical, ...rest } = metricsForNewPadding({
+            x,
+            y,
+            height,
+            width,
+            currentPadding: padding,
+            newPadding,
+          });
+          updates = {
+            ...updates,
+            ...rest,
+            padding: newPadding,
```

Comment:
What if the previous padding was already higher than the `DEFAULT_PADDING_FOR_MODE`? It should probably remain, I suppose?

Reply:
I'm actually wondering if this should change the padding prop at all or should it just be default padding that's always added?

Currently, the user could just change the padding back to 0 and there would be no space anymore -- is this the expected behavior?

Reply:
@carlos-kelly Could you confirm if it's intended behavior that the user is able to change the padding lower than the default padding style?
E.g. if the default is 6px for Fill, the user can still change it to `1px` -- should this be allowed?

Also, if the padding is set to something higher, e.g. the padding is 20px and the user changes the background mode from Highlight to Fill, it's changed back to 6px -- even though the original padding is higher than 6px -- shouldn't it just stay at 20px then?

Reply:
That is a good product question I am unsure of. Let me ask.